test/sslapitest.c

```diff
@@ -225,133 +190,103 @@ static int test_keylog_output(char *buffer, const SSL *ssl,
                 server_handshake_secret_count++;
             else if (strcmp(token, "CLIENT_TRAFFIC_SECRET_0") == 0)
                 client_application_secret_count++;
-            else if (strcmp(token, "SERVER_TRAFFIC_SECRET_0") == 0)
+            else
                 server_application_secret_count++;
 
             client_random_size = SSL_get_client_random(ssl,
                                                        actual_client_random,
                                                        SSL3_RANDOM_SIZE);
-            if (client_random_size != SSL3_RANDOM_SIZE) {
-                printf("Unexpected short client random.\n");
+            if (!TEST_size_t_eq(client_random_size, SSL3_RANDOM_SIZE))
                 return 0;
-            }
 
-            token = strtok(NULL, " \n");
-            if (!token) {
-                printf("Unexpected short client handshake secret log.\n");
+            if (!TEST_ptr(token = strtok(NULL, " \n")))
                 return 0;
-            }
-            if (strlen(token) != 64) {
-                printf("Bad value for client random: %s\n", token);
+            if (!TEST_size_t_eq(strlen(token), 64))
                 return 0;
-            }
-            if (compare_hex_encoded_buffer(token, 64, actual_client_random,
-                                           client_random_size)) {
-                printf("Bad value for client random: %s\n", token);
+            if (!TEST_false(compare_hex_encoded_buffer(token, 64,
+                                           actual_client_random,
+                                           client_random_size)))
                 return 0;
-            }
 
-            token = strtok(NULL, " \n");
-            if (!token) {
-                printf("Unexpectedly short master secret log.\n");
+            if (!TEST_ptr(token = strtok(NULL, " \n")))
                 return 0;
-            }
 
             /*
              * TODO(TLS1.3): test that application traffic secrets are what
              * we expect */
         } else {
-            printf("Unexpected token in buffer: %s\n", token);
+            TEST_info("Unexpected token %s\n", token);
             return 0;
         }
 
         token = strtok(NULL, " \n");
```

Comment:
Since you made a for loop of the while, maybe this should move to the "after" part of the `for` statement?